### PR TITLE
[JENKINS-72407] missing folder icon

### PR DESCRIPTION
**Pull request by mawinter69**
when a folder icon is provided and relies on the getImageOf method this leads to a missing icon, e.g. gitlab branch source or when using a custom folder icon
So fall back to the ballColorTd when we have no iconname which handles all cases properly.

See [JENKINS-72407](https://issues.jenkins.io/browse/JENKINS-72407).

### Testing done
- Install custom-folder-icon plugin,
- create folder and configure it to use a custom icon
- go to jenkins root dashboard, the icon should be properly shown

### Proposed changelog entries

- JENKINS-72407, fix missing folder icons

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by Pldi23 — APPROVED** [review]
I manually tested use cases reported in https://issues.jenkins.io/browse/JENKINS-72576, the fix works fine. Thank you.

**Review by amuniz — APPROVED** [review]

**Review by Vlatombe — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Comment by aneveux** [comment]
Hello :wave:

I see in the [JENKINS-72407](https://issues.jenkins.io/browse/JENKINS-72407) jira ticket that this issue is labeled with `lts-candidate`. Does this mean that this Pull Request will be backported to a `2.440.1` version of Jenkins for the February release?

ping @MarkEWaite who merged this Pull Request.

Thanks a lot for the info!

**Comment by MarkEWaite** [comment]
> I see in the [JENKINS-72407](https://issues.jenkins.io/browse/JENKINS-72407) jira ticket that this issue is labeled with `lts-candidate`. Does this mean that this Pull Request will be backported to a `2.440.1` version of Jenkins for the February release?

Thanks for asking.  Yes, the intent is that it should be backported to 2.440.1 along with the security fixes.  

The Jenkins 2.440.1 LTS release checklist is available as:

* https://github.com/jenkins-infra/release/issues/497

**Comment by aneveux** [comment]
Awesome! Thanks a lot @MarkEWaite :heart:

**Comment by MarkEWaite** [comment]
@aneveux I'd love it if you could review the other pending pull request.

* https://github.com/jenkinsci/jenkins/pull/8832

**Pull request merged**
